### ISSUE TITLE
Measuring the electricity consumption of indiv. items and actions @home

### ISSUE THREAD

**Issue by iteles** [dwyl/home#134]
## Why?
Monitoring electricity consumption allows us to:
+ Better understand and quantify the consumption of each electrical appliance @home, many of which are very common, everyday appliances that lots of people will have in their own homes (a simple 7-9kg washing machine for example)
+ Better understand and quantify how our _actions_ affect electrical consumption (leaving the heating on when we leave the house so it's toasty when we get back, or using the washing machine on a 'Cotton' load vs a 'short cycle')
+ Better understand and quantify how leaving things on standby affects consumption (leaving chargers plugged in even if they aren't charging anything for example)
+ Better understand and quantify _when_ in the day the most energy is being used and whether that is the optimal time to be doing so

But ***why*** is knowing about our electrical consumption important? The additional awareness allows us to make a **conscious choice** on where we can either _cut_ our energy consumption or use it wisely. 
But why do _this_?

+ Chief among the reasons is a **reduction in carbon emissions**, both directly and _indirectly_
  + This is why we have also opted for [solar panels](#20) 
  + Whilst _our_ energy has been coming solely from renewable resources (the image below comes directly from our electricity bill, showing a 100% hydropower consumption), this is not the case for _many_ people (see below - the average Portuguese household generates over half a ton of CO2 from their energy consumption alone every year) and **one of the key tenets of @home is to provide learnings that people can take away with them**. This is the place to test things like this out!
  + For those who cannot _afford_ to switch to a **100% renewable energy supplier** (we used Ecotricity in the UK for example and know first hand that they are marginally more expensive than certain other electricity providers), cutting consumption is a **cost-free way to contribute to betterment of the environment**
![image](https://user-images.githubusercontent.com/4185328/88478035-ee3a3180-cf3c-11ea-97f4-34335e37c95a.png)
  + *Reducing carbon emissions* is _critical_ to slow down climate change, help sustain the finite resources we already have and result in **cleaner air**
+ I would hazard a guess that **cost savings** are either number 1 or number 2 on _everyone's_ list here
  + Cutting energy consumption leads to a direct cost saving in energy costs 
  + In a similar vein, understanding _when_ in the day you are consuming most energy also allows you to shift certain actions to times where your energy costs are lower
+ If you have solar panels, understanding peak consumption vs production times also allows you to **make the most of your renewable resources**
+ One of the [UN's Sustainable Goals for Global Development](https://www.un.org/sustainabledevelopment/) (to be reached by 2030) is [sustainable consumption and production](https://www.un.org/sustainabledevelopment/sustainable-consumption-production/) of multiple resources, with _energy_ among them
  + Goal target **12.8** specifically reads: "By 2030, ensure that people everywhere have the relevant information and awareness for sustainable development and lifestyles in harmony with nature"

## What?
Monitoring our energy consumption and knowing the daily, weekly and monthly consumption both in summer and in winter as well as the _individual_ consumption of individual appliances and actions.
These should be published for everyone to be encouraged to use this data, understand our actions (e.g. why we run pool filters at peak sunshine hours) and be encourage to seek out ways to reduce their energy consumption in their own homes or places they stay at next.
This is great not just for the environment but for cost savings.

## How?
More than 10 years ago, we had an energy monitor that we could clip around the main electrical cable coming out of our electricity meter and it would tell us the exact consumption at that point in time in kW as well as give us monthly totals and averages.

This was a fantastic way to keep track of our energy consumption and gave us real insight into the consumption of each appliance as well as the difference our actions made.

The _how_ is exactly what this issue was opened to research 🚀

**Comment by nelsonic** [comment]
@iteles I created a "Research" task: #144 and assigned it to @th0mas  
I know you had done a bit of research into this. 🔍 (_what did you conclude_?)
I feel that it's a good task for Tom because he is UK-based which means any products ordered online for testing will arrive _much_ faster and he can do the software integration work too. 👨‍💻 
Ultimately we want to have a real-time dashboard telling us _exactly_ how much Energy we are using. That's a hardware+software challenge so the approach needs to be integrated, hence #144. 👍

**Comment by iteles** [comment]
@nelsonic I was in the initial stages of the research so more generic/informative for anyone who wanted to know what was possible (and certainly hadn't gone as far as researching open source solutions yet, my order was 'what's available > what's available in Portugal > Open source solutions), but to summarise my open tabs:

There are 3 possible ways of measuring:
## 1. Whole House energy monitors 
These are the quite widely available now (when we got our first one in... 2008? there were very few choices!).

They consist of devices that either connect directly _inside_ your electrical panel like [Sense](https://sense.com/)
![image](https://user-images.githubusercontent.com/4185328/92989586-30a7d580-f4cd-11ea-934b-e0ecbc3625f8.png)

Or (most) simply clip onto the main wiring that leaves your electrical cupboard like [Loop Energy Saver](https://loop.homes/product/loop-your-essential-energy-saving-assistant/) (either inside it or just outside):
![image](https://user-images.githubusercontent.com/4185328/92989703-0571b600-f4ce-11ea-9d64-ce7ef70c0817.png)

These usually come with apps (both desktop and smartphone) that provide features such as: 
+ Give you realtime insights into the total energy consumption of your home
+ Track and graph your usage over time
+ Allow you to understand your base usage (i.e what your home consumes when you think 'everything is off'
+ Most also offer energy-saving tips (not specific to your household)

The more advanced apps also offer:
+ Ability to see what your monthly bill is expected to be 
+ Ability to [switch your energy provider](https://loop.homes/switch/)
+ Help you understand **exactly which appliance is on** through use of electronic signatures and machine learning
  + This is still very embryonic and the best I found in my research was [Sense](https://blog.sense.com/articles/how-does-sense-detect-my-devices/), although there are still quite a few [limitations](https://blog.sense.com/10-steps-to-get-started-with-sense/) - such as taking 12 months to detect about 25 devices in your home


##2. Individual appliance smart plugs/plug power monitors

Probably still be good to have one of these to double-check individual appliances that can be plugged in, especially in a house with a lot of people where it's harder to control that no one is switching something else on just as you're trying to measure the electrical requirements of one particular appliance.

https://www.trotec24.pt/maquinas/acessorios-gerais/acessorios-maquinas/medidor-de-consumo-eletrico-bx11-f-versao-francesa.html
![image](https://user-images.githubusercontent.com/4185328/92948910-154bb480-f452-11ea-9e6f-04e243644d79.png)

There are a number of more specialised and expensive smart plugs, like the Elgato Eve, which works with the Apple HomeKit (I imagine we could put one of these together ourselves using OSS and hardware components if we wanted to):
https://www.amazon.co.uk/Eve-Energy-connected-necessary-Bluetooth/dp/B01KHLWAHM 
![image](https://user-images.githubusercontent.com/4185328/92951750-d0764c80-f456-11ea-8246-e27a69beaf18.png)


## 3. Smart energy apps
These are the lowest rung of the ladder in that they are simply apps that rely on other external equipment to do the measuring without a hardware solution of their own. They try to use the data from either smart meters or from smart plugs to patch together an overall view and are _much_ less accurate.

These also tend to be quite market-specific because of how they hook into a country or company's specific smart meters and they're also not very sophisticated - an example is this [Energy Consumption Analyzer](https://play.google.com/store/apps/details?id=at.topfen.ecas&hl=en).

Not relevant for what we're looking for.


# Available in Portugal
I was curious to learn about restrictions in using the equipment across multiple countries (things we might need to pay attention to if we were building our own solution) so I dropped an email to Loop Energy to find out if their products could be used in Portugal.

I got an interesting response:
![image](https://user-images.githubusercontent.com/4185328/92990813-55a14600-f4d7-11ea-9c31-b60ee0c8fe38.png)

This was good news as there's nothing keeping us from having our own OSS solution.
But I was curious to know what was already on the market in Portugal:
+ [BeEnergy from Infocontrol](https://www.infocontrol.pt/gestao-de-energia-e-eficiencia-energetica/monitorizacao-consumos-energia/beenergy/) - a software-only home monitoring system with the ability to define tariffs (and therefore calculate bills), compatible with a couple of hardware solutions to pull in data
+ [Wibeee Home](https://wibeee.com/en/wibeee-home/) - Spanish company providing both software and hardware solutions (both plugs _and_ full home monitoring system with clips) for energy monitoring, relatively advanced technically with an API allowing for multiple home management. _They also have a decent free plan_ so clearly the value of their business is in the data they collect (and somewhat in hardware)
![image](https://user-images.githubusercontent.com/4185328/92991084-fe03da00-f4d8-11ea-9dcd-1d9c6f952885.png)
+ [Bee2Energy](https://www.ceb-solutions.com/pt-pt/produtos/bee2energy/) - a Portuguese company which offers monitoring software for a large range of industries under the 'intelligent cities' umbrella. Their product is software-only but compatible with multiple hardware solutions, includes tariff and forecasting functionality
![image](https://user-images.githubusercontent.com/4185328/92991265-ab2b2200-f4da-11ea-92d2-6ead50f2bed4.png)

+ [PierEnergy](
https://pierenergy.pt/monitorizacao-de-consumos/) - this is a very opaque website, they seem to offer some kind of energy monitoring solution but they don't tell us which one 🤷‍♀️ 


In short, everything is very much closed source and the cost of the end of the month bill seems to be an integral feature in each solution, more so than in the international options I've seen.

**Comment by iteles** [comment]
@nelsonic Now that the Why, What and some initial, more general research has been captured above, I think we can close this issue entirely in favour of #144 👍 

Agreed?

**Comment by nelsonic** [comment]
@iteles I don't think we can close this issue until it is captured in a Blog Post. 💭

**Comment by iteles** [comment]
@nelsonic Is there anything you reckon we _need_ from this shop?
https://www.efectoled.com/pt/703-comprar-smart-home

Here is my current shopping cart just over €3 short of free postage 😉 :
![image](https://user-images.githubusercontent.com/4185328/97903211-9464b700-1d36-11eb-8364-b704639db53e.png)

https://www.efectoled.com/pt/comprar-medidores-de-energia/1777-temporizador-digital-8445044009848.html
https://www.efectoled.com/pt/comprar-temporizadores-programadores/1753-temporizador-analogico-compacto-con-enchufe.html

I was thinking of something like this for temporarily having by our front door at night:
https://www.efectoled.com/pt/comprar-alarmes/1897-timbre-para-puertas-8445044010790.html

**Comment by nelsonic** [comment]
@iteles I would just add another **Medidor controlador** to reach the "free" postage threshold. 🔌 
The front door motion detecting chime is more of an annoyance than anything else. 💭

**Comment by iteles** [comment]
Ok, will get these today 👍🏻
